server/graphql/loaders/expenses.ts

```diff
@@ -72,3 +80,79 @@ export const attachedFiles = (): DataLoader<number, ExpenseAttachedFile[]> => {
     return sortResultsArray(expenseIds, attachedFiles, file => file.ExpenseId);
   });
 };
+
+/**
+ * Expense loader to check if userTaxForm is required before expense payment
+ */
+export const userTaxFormRequiredBeforePayment = (req): DataLoader<number, boolean> => {
+  return new DataLoader(async (expenseIds: number[]) => {
+    const results = await sequelize.query(`
+      SELECT e.id "expenseId", e."UserId" "userId", ld."requestStatus" "legalDocRequestStatus", d."documentType" "requiredDocument",
+        SUM (amount) AS total
+      FROM "Expenses" e
+      INNER JOIN "Collectives" c ON c.id = e."CollectiveId"
+      LEFT JOIN "RequiredLegalDocuments" d ON d."HostCollectiveId" = c."HostCollectiveId"
+                                              AND d."documentType" = 'US_TAX_FORM'
+      LEFT JOIN "LegalDocuments" ld ON ld."CollectiveId" = e."FromCollectiveId"
+                                        AND ld.year = date_part('year', e."incurredAt")
+                                        AND ld."documentType" = 'US_TAX_FORM'
+      WHERE e.status IN ('PENDING', 'APPROVED', 'PAID', 'PROCESSING')
+      AND e."UserId" IN (SELECT "UserId" FROM "Expenses" WHERE "Expenses".id IN (:expenseIds))
+      AND e.type NOT IN ('RECEIPT')
+      AND "incurredAt" BETWEEN e."incurredAt" AND (e."incurredAt" + interval '1 year')
+      GROUP BY e.id, e."UserId", d."documentType", ld."requestStatus"
+    `, {
+       type: sequelize.QueryTypes.SELECT,
+       model: models.Expense,
+       mapToModel: true,
```

Comment:
Why do we need to map to model? Wouldn't it be simple to have `raw: true` so that we can access values directly without having to go through `dataValues`?